### PR TITLE
Fix for `NSInvalidArgumentException` in `GetPromotionalOffer`

### PR DESCRIPTION
**Pull request by vegaro**
We were not handling errors in `GetPromotionalOffer` and we were trying to send a nil response to the Unity message layer

[TRIAGE-276]

[TRIAGE-276]: https://revenuecats.atlassian.net/browse/TRIAGE-276?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: RevenueCat/Plugins/iOS/PurchasesUnityHelper.m [hunk 1]
```diff
@@ -265,7 +265,15 @@ - (void)promotionalOfferForProductIdentifier:(NSString *)productIdentifier
...
                                                       discount:discountIdentifier
                                                completionBlock:^(NSDictionary *_Nullable responseDictionary, RCErrorContainer *_Nullable error) {
-        [self sendJSONObject:responseDictionary toMethod:GET_PROMOTIONAL_OFFER];
+        NSMutableDictionary *response;
+        if (error) {
+            response = [NSMutableDictionary new];
+            response[@"error"] = error.info;
+
+        } else {
+            response = [NSMutableDictionary dictionaryWithDictionary:responseDictionary];
```
Comment (joshdholtz) [inline RevenueCat/Plugins/iOS/PurchasesUnityHelper.m:274]: Do we need to make sure that `responseDictionary` isn't `nil` at all? Or is it safe to assume it never will be?

**Review by joshdholtz — APPROVED** [review]
One small question but looks good!

File: RevenueCat/Plugins/iOS/PurchasesUnityHelper.m (hunk 1 above)
Comment (NachoSoto) [reply RevenueCat/Plugins/iOS/PurchasesUnityHelper.m:274]: Yeah I'd create the `NSMutableDictionary` in line 268 so we don't risk sending `nil`.

File: RevenueCat/Plugins/iOS/PurchasesUnityHelper.m [hunk 2]
```diff
@@ -265,7 +265,15 @@ - (void)promotionalOfferForProductIdentifier:(NSString *)productIdentifier
...
+        }
+        [self sendJSONObject:response toMethod:GET_PROMOTIONAL_OFFER];
```
Comment (NachoSoto) [inline RevenueCat/Plugins/iOS/PurchasesUnityHelper.m:276]: Maybe send an immutable copy?
```suggestion
        [self sendJSONObject:response.copy toMethod:GET_PROMOTIONAL_OFFER];
```

Comment (NachoSoto) [reply RevenueCat/Plugins/iOS/PurchasesUnityHelper.m:276]: Actually I think you could simplify this while thing like this:
```objc
NSDictionary *response = (error)
? @{
  @"error": error.info
}
: responseDictionary;
```

Comment (joshdholtz) [reply RevenueCat/Plugins/iOS/PurchasesUnityHelper.m:276]: I don't understand this language

File: RevenueCat/Plugins/iOS/PurchasesUnityHelper.m (hunk 1 above)
Comment (vegaro) [reply RevenueCat/Plugins/iOS/PurchasesUnityHelper.m:274]: Initializing to an empty dictionary will break the unity code where we try to create the classes from the json response. I think we should create an error here, but the error should be in the purchases hybrid common side so all hybrids are protected from that unexpected case where both the response and the error are nil

Comment (vegaro) [reply RevenueCat/Plugins/iOS/PurchasesUnityHelper.m:274]: Oh actually, for incompatible versions phc sends nil for both 🤔

Comment (vegaro) [reply RevenueCat/Plugins/iOS/PurchasesUnityHelper.m]: I opened this PR https://github.com/RevenueCat/purchases-hybrid-common/pull/338/ to stop sending two nils, and also created an error in this PR if both responses and error are nil

**Review by joshdholtz — APPROVED** [review]
LGTM 💪

File: RevenueCat/Plugins/iOS/PurchasesUnityHelper.m (hunk 2 above)
Comment (tonidero) [reply RevenueCat/Plugins/iOS/PurchasesUnityHelper.m:281]: Hmm right, that seems to be the same it is right now (with different indentation)?

**Review by tonidero — APPROVED** [review]

**Pull request merged**